targets/simple_switch/main.cpp

```diff
@@ -20,15 +20,81 @@
 
 /* Switch instance */
 
+#ifdef BM_HAVE_DLOPEN
+#  include <dlfcn.h>
+#endif  // BM_HAVE_DLOPEN
 #include <bm/SimpleSwitch.h>
 #include <bm/bm_runtime/bm_runtime.h>
 #include <bm/bm_sim/target_parser.h>
 
+#include <sstream>
+#include <string>
+#include <vector>
+
 #include "simple_switch.h"
 
 namespace {
 SimpleSwitch *simple_switch;
-bm::TargetParserBasic *simple_switch_parser;
+
+std::string load_modules_option = "load-modules";
+
+class TargetParserModules : public bm::TargetParserBasic {
```

Comment:
Suggest renaming this to `SimpleSwitchParser` now that this class is meant to regroup all of simple_switch command-line options.

Reply:
Will do.